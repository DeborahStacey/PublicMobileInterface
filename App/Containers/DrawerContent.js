import React, { Component } from 'react'
import { ScrollView, BackAndroid, Alert } from 'react-native'
import styles from './Styles/DrawerContentStyle'
import DrawerButton from '../Components/DrawerButton'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { create } from 'apisauce'

const db = create({
  baseURL: 'https://cat.ddns.net/Backend/api.php',
  headers: {'Content-Type': 'application/json'}
})

class DrawerContent extends Component {

  componentDidMount () {
    BackAndroid.addEventListener('hardwareBackPress', () => {
      if (this.context.drawer.props.open) {
        this.toggleDrawer()
        return true
      }
      return false
    })
  }

  toggleDrawer () {
    this.context.drawer.toggle()
  }

  handlePressComponents = () => {
    this.toggleDrawer()
    NavigationActions.componentExamples()
  }

  handlePressUsage = () => {
    this.toggleDrawer()
    NavigationActions.usageExamples()
  }

  handlePressAPI = () => {
    this.toggleDrawer()
    NavigationActions.apiTesting()
  }

  handlePressTheme = () => {
    this.toggleDrawer()
    NavigationActions.theme()
  }

  handlePressDevice = () => {
    this.toggleDrawer()
    NavigationActions.deviceInfo()
  }

  handleSignOut = () => {
    Alert.alert(
      'Confirm',
      'Are you sure you want to sign out?',
      [
        {
          text: 'Sign Out',
          onPress: () => {
            db.post('/user/logout').then((response) => {
              if (response.ok) {
                NavigationActions.signIn()
                this.toggleDrawer()
                Alert.alert('Signed out successfully')
              } else {
                Alert.alert('An error has occurred')
              }
            })
          }
        },
        {
          text: 'Cancel'
        }
      ]
    )
  }

  render () {
    return (
      <ScrollView style={styles.container}>
        <DrawerButton text='Home' onPress={NavigationActions.presentationScreen} />
        <DrawerButton text='Account' onPress={NavigationActions.accountInfo} />
        <DrawerButton text='Cats' onPress={NavigationActions.catList} />
        <DrawerButton text='Sign Out' onPress={this.handleSignOut} />
      </ScrollView>
    )
  }
}

DrawerContent.contextTypes = {
  drawer: React.PropTypes.object
}

export default DrawerContent
