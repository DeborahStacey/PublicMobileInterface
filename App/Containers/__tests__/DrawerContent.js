import 'react-native'
import React from 'react'
import DrawerContent from '../DrawerContent'
import renderer from 'react-test-renderer'

jest.mock('react-native-router-flux', () => {
  return {
    Actions: {
      presentationScreen: () => {},
      accountInfo: () => {},
      catList: () => {}
    }
  }
})

it('renders correctly', () => {
  const tree = renderer.create(
    <DrawerContent />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
