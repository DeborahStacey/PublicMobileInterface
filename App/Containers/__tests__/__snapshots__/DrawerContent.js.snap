exports[`test renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
      }
    } />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      Home
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      Account
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      Cats
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      Sign Out
    </Text>
  </View>
</ScrollView>
`;
